### PR TITLE
Handle shutdown request from ControlTower

### PR DESCRIPTION
**Pull request by rickaha**
Ublox chip might need some time to save:
mUbloxRover->saveOnShutdown();
 
If this is the case, "now" can be removed from:
QProcess::startDetached("sudo", QStringList() << "shutdown" << "-r" << "now");

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -158,7 +159,20 @@ int main(int argc, char *argv[])
 
     // Perform safe shutdown
     signal(SIGINT, terminationSignalHandler);
-    QObject::connect(&a, &QCoreApplication::aboutToQuit, &mavsdkVehicleServer, &MavsdkVehicleServer::saveParametersToXmlFile);
+    QObject::connect(&a, &QCoreApplication::aboutToQuit, [&](){
+        mUbloxRover->saveOnShutdown();
+        mavsdkVehicleServer.saveParametersToXmlFile();
+    });
+    QObject::connect(&mavsdkVehicleServer, &MavsdkVehicleServer::shutdownOrRebootOnboardComputer, [&](int value){
+        qApp->quit();
```
Comment (marvind) [inline main.cpp:167]: is "mUbloxRover->saveOnShutdown();" missing here?

File: main.cpp [hunk 2]
```diff
@@ -158,7 +159,20 @@ int main(int argc, char *argv[])
...
+        if (value == 1) {
+           qDebug() << "\nSystem reboot...";
+           QProcess::startDetached("sudo", QStringList() << "shutdown" << "-r" << "now");
+        }else if (value == 2){
```
Comment (marvind) [inline main.cpp:171]: Should be bool or enum (see https://github.com/RISE-Dependable-Transport-Systems/WayWise/pull/30/files#r1115310881).

**Review by marvind — APPROVED** [review]
Looks good! OK to merge once changes in WayWise that this depends on are merged.

File: main.cpp (hunk 2 above)
Comment (rickaha) [reply main.cpp]: Fixed

**Pull request merged**